jctools-core/src/main/java/org/jctools/util/ThreadHints.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright 2016 Gil Tene
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jctools.util;
+
+@InternalAPI
+public final class ThreadHints
+{
+
+    /**
+     * Set this system property to true to disable {@link #onSpinWait()}.
+     */
+    public static final String DISABLE_ON_SPIN_WAIT = "jctools.disable.onSpinWait";
+    private static final boolean okToInvokeInternalThreadHintsCaller;
+
+    static
+    {
+        boolean okToInvokeMHCaller = false;
+        if (!Boolean.getBoolean(DISABLE_ON_SPIN_WAIT))
+        {
+            try
+            {
+                Class.forName("java.lang.invoke.MethodHandle");
+                okToInvokeMHCaller = true;
+            }
+            catch (Exception e)
+            {
+                okToInvokeMHCaller = false;
+            }
+        }
+        // okToInvokeThreadHintsMHCaller will be false for e.g. Java SE 6 and earlier:
+        okToInvokeInternalThreadHintsCaller = okToInvokeMHCaller;
+    }
+
+    // prevent construction...
+    private ThreadHints()
```

Comment:
I used to do this too, but my current thinking is that it is pointless. The class is final, it has no fields, no methods that are not static. If some idiot wants to instantiate it, so what?